File: server/router/lg/apply.go
package lg

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ApplyRouter struct {
}

func (s *ApplyRouter) InitApplyRouter(Router *gin.RouterGroup) {
	applyRouter := Router.Group("apply").Use(middleware.OperationRecord())
	applyRouterWithoutRecord := Router.Group("apply")
	var applyApi = v1.ApiGroupApp.LgApiGroup.ApplyApi
	{
		applyRouter.POST("createApply", applyApi.CreateApply)             // 新建Apply
		applyRouter.DELETE("deleteApply", applyApi.DeleteApply)           // 删除Apply
		applyRouter.DELETE("deleteApplyByIds", applyApi.DeleteApplyByIds) // 批量删除Apply
		applyRouter.PUT("updateApply", applyApi.UpdateApply)              // 更新Apply
	}
	{
		applyRouterWithoutRecord.GET("findApply", applyApi.FindApply)       // 根据ID获取Apply
		applyRouterWithoutRecord.GET("getApplyList", applyApi.GetApplyList) // 获取Apply列表
	}
}

File: web/src/api/lg/apply.js
import service from '@/utils/request'

export const createApply = (data) => {
  return service({
    url: '/apply/createApply',
    method: 'post',
    data
  })
}

export const deleteApply = (data) => {
  return service({
    url: '/apply/deleteApply',
    method: 'delete',
    data
  })
}

export const deleteApplyByIds = (data) => {
  return service({
    url: '/apply/deleteApplyByIds',
    method: 'delete',
    data
  })
}

export const updateApply = (data) => {
  return service({
    url: '/apply/updateApply',
    method: 'put',
    data
  })
}

export const findApply = (params) => {
  return service({
    url: '/apply/findApply',
    method: 'get',
    params
  })
}

export const getApplyList = (params) => {
  return service({
    url: '/apply/getApplyList',
    method: 'get',
    params
  })
}
